### ISSUE TITLE
Is it hard to add browser support?

### ISSUE THREAD

**Issue by bigfarofa** [jrop/nedb-promise#11]
Before anything else I just want to thank you for the wrapper. I'm interested in using  the browser version of nedb `nedb/browser-version/out/nedb.min.js`, but as it seems the wrapper does not support it.
In theory there are two options:
    1 - Create a copy of the wrapper, but insted of importing `nedb`, importing `nedb/browser-version/out/nedb.min.js`
     2 - Create a exclusive option like `_browser`. If this option is enabled, the wrapper imports the browser version instead of node's.

**Comment by bigfarofa** [comment]
I forgot to check the method `fromInstance`. Probably it will work.

**Comment by jrop** [comment]
@DREZZ3R `fromInstance` will probably work.  However, at the top of `index.js`, I have the line `require('nedb')`, so this will probably require a bit of refactoring to fully support the browser.  I can probably accomplish this with the following steps:

1. Split `fromInstance` into its own file
2. Create `browser.js` that requires `nedb/browser-version/out/nedb.min.js` and then uses `fromInstance` to promisify that
3. Create a WebPack config to bundle all of this together

It would also be good to add some tests that run in the browser, using Karma, or something like that.  Perhaps I'll take the opportunity to also convert this repo to TypeScript

**Comment by bigfarofa** [comment]
@jrop I've tried using the browser version, but it just does not work. I tested with this code:
```javascript
import Nedb from 'nedb/browser-version/out/nedb.min.js';
import nedbPromify from 'nedb-promise';
let db = nedbPromify.fromInstance(new Nedb('images', {autoload: true}))
db.findOne({})
.then(("findOne RES", res) => console.log(res))
.catch(("findOne ERR", err) => console.log(err))
```
This does not give any error, it just stops. If I use this in an `async` function everything after the query, is not executed.

**Comment by jrop** [comment]
@DREZZ3R That does not seem to be valid syntax:

```js
.then(("findOne RES", res) => console.log(res))
// Should be?: .then((res) => console.log("findOne RES", res))
.catch(("findOne ERR", err) => console.log(err))
// Should be?: .catch((err) => console.log("findOne ERR", err))
```

It seems that the only parsers that accept that syntax are Flow and TypeScript, and even then, they see that as a parenthesized comma-expression, with a second argument of `console.log(...)`.  I bet if you changed to the syntax proposed above the callbacks might get executed

**Comment by bigfarofa** [comment]
@jrop  I figured it out.  Yes, the syntax I wrote here is wrong, but in the app's source code was right and didn't work.
The problem was how I created the Nedb instance.

<b>How it was:</b>
```javascript
let db = nedbPromify.fromInstance(new Nedb('images', {autoload: true}))
```
<b>How it should've been:</b> 
```javascript 
let db = nedbPromify.fromInstance(new Nedb({filename: "images", autoload: true}))
```
I thought the first argument was the DB name and then options as second. Nedb only accepts one argument either a string or a object with options. I feel stupid, thank you for your patience.
